pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -92,6 +94,34 @@ public static PinotQuery compileToPinotQuery(String sql)
     return pinotQuery;
   }
 
+  static void validate(PinotQuery pinotQuery)
+      throws SqlCompilationException {
+    // Sanity check group by query: All identifiers in selection list should be also included in group by list.
+    if (pinotQuery.getGroupByList() != null) {
```

Comment:
NIT: use if(pinotQuery.getGroupByList() == null) return; to reduce the nest levels.

Reply:
done